db scheme added

AlexanderSharygin opened the pull request.

AlexanderSharygin committed:
db scheme added

kskripkin commented:
Привет. Отличная работа)
Но есть несколько вопросов.
1. Не вижу первичных ключей в таблицах, они есть? Вроде они должны ключиком обозначаться на схеме.
2. Поле id в таблице users я бы переименовал в user_id для упрощения чтения запросов (была рекомендация на вебинаре про проектированию БД).
3. В films аналогично пункту 2.
4. В таблице films столбец duration и rate_id точно нужен нам формата bigint? Учитывая что продолжительность фильма мы указываем в минутах, а рейтинг измеряется у нас в количестве менее 10 пунктов (по условию задачи спринта 11).

В целом, я бы не плодил столько таблиц так как это усложняет работу с ними, занимает больше места на диске, а в случае изменения той же таблицы genres или genres_films лок упадет и на таблицу films. Но тут, конечно, на твое усмотрение)

AlexanderSharygin commented:
Спасибо за похвалу и замечания
1. Первичные ключи есть - поля являющиеся PK выделены полужирным (фича редактора в котором рисовал схему)
2. fixed
3. fixed
4. fixed

По количеству таблиц - я старался делать все под форму 3 NF но конечно я могу быть не прав - но поглядим что скажут профессионалы в 12 финальном задании))

AlexanderSharygin committed:
Update db_cheme.jpg

kskripkin commented:
Все отлично. Спасибо.
